apiserver/charms.go

```diff
@@ -45,14 +44,16 @@ func (h *charmsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	var err error
 	switch r.Method {
 	case "POST":
-		err = h.servePost(w, r)
+		err = errors.Annotate(h.servePost(w, r), "cannot upload charm")
```

Comment:
if it were me, I'd split this into two lines... the error handling here obscures what's actually being done, IMO... but that's mostly trivial.